src/AbstractSingleInstancePluginManager.php

```diff
@@ -26,14 +26,6 @@ abstract class AbstractSingleInstancePluginManager extends AbstractPluginManager
      */
     protected string $instanceOf;
 
-    /**
```

Comment:
Removing type-narrowing here as it actually breaks psalms conditional detection.
I do not really have time to improve this right now and thus removing this for now.
If someone still manages to return invalid stuff, that would be on purpose and thus we should not really care about it.